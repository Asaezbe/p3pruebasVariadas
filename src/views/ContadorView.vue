<template>
  
  <h2 class="titulo-principal">{{ tituloPorDefecto }}</h2>
  <p class="counter">{{ contador }}</p>

  <div>
    
    <button class="btn primary incrementar" v-on:click="incrementar">+1</button>
    <button class="btn warning disminuir" v-on:click="disminuir">-1</button>
   
  </div>
</template>

 


<script>
import Contador from '../../store/index';  // Importa el script desde la nueva ruta

export default {
  name: 'ContadorView',
  mixins: [Contador],  // Mezcla el componente Contador en este componente
};
</script>




<style>
.titulo-principal {
  font-weight: 500;
  color: blueviolet;
}

.btn {
  border-radius: 1rem;
  margin: 0 5px;
  width: 100px;
  padding: 5px;
  font-size: 1rem;
}
.primary {
  background-color: rgb(77, 77, 231);
  color: rgb(255, 255, 255);
}
.counter {
  font-size: 36px;
}

.warning {
  background-color: rgb(230, 53, 53);
  color: rgb(255, 255, 255);
}
</style>