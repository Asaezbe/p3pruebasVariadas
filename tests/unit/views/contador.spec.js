import { shallowMount } from '@vue/test-utils';
import ContadorView from '@/views/ContadorView.vue';

describe('Componente ContadorView.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ContadorView, {
      propsData: {
        titulo: 'Mi Contador',
        inicio: 0
      }
    });
  });

  test('Validación de match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('El título debe ser "Mi Contador"', () => {
    const titulo = wrapper.find('.titulo-principal');
    expect(titulo.text()).toBe('Mi Contador');
  });

  test('El contador inicia en el valor correcto', () => {
    const counter = wrapper.find('.counter');
    expect(counter.text()).toBe('0'); // Valor inicial es 1
  });

  test('Incrementar el contador', async () => {
    const botonIncrementar = wrapper.find('.incrementar');
    await botonIncrementar.trigger('click'); // Simula el clic en el botón
    const counter = wrapper.find('.counter');
    expect(counter.text()).toBe('1'); // Después de incrementar, debe ser 1
  });

  test('Disminuir el contador', async () => {
    const botonDisminuir = wrapper.find('.disminuir');
    await botonDisminuir.trigger('click'); // Simula el clic en el botón
    const counter = wrapper.find('.counter');
    expect(counter.text()).toBe('-1'); // Después de disminuir, debe ser 0 (suponiendo que no hay lógica para impedirlo)
  });

 
});
