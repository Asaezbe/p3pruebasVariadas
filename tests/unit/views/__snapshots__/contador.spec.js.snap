// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente ContadorView.vue Validaci√≥n de match con el snapshot 1`] = `
<h2 class="titulo-principal">Mi Contador</h2>
<p class="counter">0</p>
<div><button class="btn primary incrementar">+1</button><button class="btn warning disminuir">-1</button></div>
`;
